bindings/python/PySound.cpp

```diff
@@ -1842,7 +1844,117 @@ Sound_timeStretchPitchScale(Sound* self, PyObject* args)
+	{
+		PyErr_SetString(PyExc_ValueError, "fps must be greater 0!");
+		return nullptr;
+	}
+
+	if(quality < 0 || quality > 2)
+	{
+		PyErr_WarnEx(PyExc_UserWarning, "Invalid quality value: using default (0 = HIGH)", 1);
+		quality = 0;
+	}
+
+	PyTypeObject* type = Py_TYPE(self);
+	Sound* parent = (Sound*) type->tp_alloc(type, 0);
+
+	if(parent != nullptr)
+	{
+		try
+		{
+			parent->sound = new std::shared_ptr<ISound>(new AnimateableTimeStretchPitchScale(*reinterpret_cast<std::shared_ptr<ISound>*>(self->sound), time_ratio, pitch_scale,
+			                                                                                 static_cast<StretcherQuality>(quality), preserve_formant != 0, fps));
+		}
+		catch(Exception& e)
+		{
+			Py_DECREF(parent);
+			PyErr_SetString(AUDError, e.what());
+			return nullptr;
+		}
+	}
+
+	return (PyObject*) parent;
+}
+
+static PyObject * 
+Sound_animateableTimeStretchPitchScale_getAnimProperty(Sound* self, PyObject* args)
+{
+    int type;
+    if (!PyArg_ParseTuple(args, "i:getAnimProperty", &type))
+        return nullptr;
+
+		auto atps = std::dynamic_pointer_cast<AnimateableTimeStretchPitchScale>(
+				*reinterpret_cast<std::shared_ptr<ISound>*>(self->sound)
+		);
+
+    if (!atps) {
+        PyErr_SetString(PyExc_TypeError, "Sound is not an AnimateableTimeStretchPitchScale!");
+        return nullptr;
+    }
+
+    AnimateablePropertyP* prop =
+        (AnimateablePropertyP*)AnimateableProperty_empty();
+    try
+    {
+        auto animProp = atps->getAnimProperty(static_cast<AnimateablePropertyType>(type));
+
+        if (!animProp) {
+            Py_DECREF(prop);
+            PyErr_SetString(PyExc_ValueError, "Invalid AnimateablePropertyType");
+            return nullptr;
+        }
+
+        prop->animateableProperty = new std::shared_ptr<AnimateableProperty>(animProp);
+
+        if (prop->animateableProperty == nullptr) {
+            Py_DECREF(prop);
+            PyErr_SetString(PyExc_ValueError, "Invalid AnimateablePropertyType");
+            return nullptr;
+        }
+    }
+    catch (const Exception& e)
+    {
+        Py_DECREF(prop);
+        PyErr_SetString(AUDError, e.what());
+        return nullptr;
+    }
+
+    return (PyObject*)prop;
+}
+
+PyDoc_STRVAR(M_aud_Sound_animateableTimeStretchPitchScale_getAnimProperty_doc, ".. method:: animateableTimeStretchPitchScale_getAnimProperty(type)\n\n"
+                                                            "   Retrieves one of the animated properties of the sound (time-stretch or pitch-scaling).\n\n"
+                                                            "   :arg type: Which animated property to retrieve (AUD_AP_*).\n"
+                                                            "   :type type: int\n"
+                                                            "   :return: The created :class:`AnimateablePropertyP` object.\n"
+                                                            "   :rtype: :class:`AnimateablePropertyP`\n");
```

Comment:
As you already noticed, the `animateableTimeStretchPitchScale_getAnimProperty` method doesn't really fit with the rest of the API. I think this should be changed so that two AnimateableProperty objects can be passed in to the `Sound_animateableTimeStretchPitchScale` method and the getter should be removed in order to fit with how everything else works here. You could even add such a constructor to the `AnimateableTimeStretchPitchScale` class.